pkg/authorizationserver/oauth2.go

```diff
@@ -0,0 +1,138 @@
+package authorizationserver
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/ardanlabs/conf"
+	"github.com/ory/fosite"
+	"github.com/ory/fosite/compose"
+	"github.com/ory/fosite/handler/openid"
+	"github.com/ory/fosite/storage"
+	"github.com/ory/fosite/token/jwt"
+	"github.com/tbd54566975/ssi-service/config"
+	"github.com/tbd54566975/ssi-service/pkg/server/framework"
+	"github.com/tbd54566975/ssi-service/pkg/server/middleware"
+)
+
+// fosite requires four parameters for the server to get up and running:
+//  1. config - for any enforcement you may desire, you can do this using `compose.Config`. You like PKCE, enforce it!
+//  2. store - no auth service is generally useful unless it can remember clients and users.
+//     fosite is incredibly composable, and the store parameter enables you to build and BYODb (Bring Your Own Database)
+//  3. secret - required for code, access and refresh token generation.
+//  4. privateKey - required for id/jwt token generation.
+var (
+	// Check the api documentation of `compose.Config` for further configuration options.
+	fositeConfig = &fosite.Config{
+		AccessTokenLifespan: time.Minute * 30,
+		GlobalSecret:        secret,
+		// ...
```

Comment:
Is this an official oatuh2.go file so you left as is? Otherwise can remove this comment